rllib/agents/ddpg/ddpg_torch_policy.py

```diff
@@ -21,8 +22,8 @@ def build_ddpg_models_and_action_dist(policy, obs_space, action_space, config):
     # TODO(sven): Unify this once we generically support creating more than
     #  one Model per policy. Note: Device placement is done automatically
     #  already for `policy.model` (but not for the target model).
-    device = (torch.device("cuda")
-              if torch.cuda.is_available() else torch.device("cpu"))
+    device = (torch.device("cuda") if torch.cuda.is_available()
+              and ray.get_gpu_ids() else torch.device("cpu"))
```

Comment:
On my computer, `ray.get_gpu_ids()` is 0, even if the number of gpu is higher than 0. It is non-zero only if a use the decorator `@remote(num_gpus=N)`, with N > 0.

Reply:
If I am not wrong, inside a trainer class if you set up the config with the GPU parameter and `ray.init` with gpu > 0, it will return true.

Reply:
Also that's how rllib decides to move the main model to the GPU.

Reply:
Yes, you have to initialize ray with num_gpus > 0.